### PR TITLE
Rough draft web RL

### PR DESCRIPTION
**Pull request by jacmantooth**

### CODE REVIEW

**Comment by jacmantooth** [comment]
Hey lars here is our website

File: content/projects/2022-rl-vs-oc/index.md [hunk 1]
```diff
@@ -1,9 +1,118 @@
...
+weight : 20
+summary: 'In recent years, reinforcement learning (RL) methods have shown success in solving optimal control problems involving ordinary differential equations. RL approaches differ drastically from traditional optimal control methods. Where prior methods have relied on the ODE models, RL tries to learn the optimal control simply by observing actions and rewards. There is tremendous activity in the sciences and engineering utilizing both methods. However, RL and optimal control theory is rarely compared on the same problem. This paper fills that gap and tries to investigate the trade-off between data-driven RL methods and model-driven optimal control methods on the continuous mountain car problem. We find that...'
```
Comment (lruthotto) [inline content/projects/2022-rl-vs-oc/index.md:6]: Instead of 'this paper' say something like in this 'REU project'?

Comment (lruthotto) [reply content/projects/2022-rl-vs-oc/index.md:6]: Also, last sentence to be completed.

File: content/projects/2022-rl-vs-oc/index.md [hunk 2]
```diff
@@ -1,9 +1,118 @@
...
 tags: ["Summer 2022"]
 ---
+
+## What is the best way to get a car out of the bottom of a hill?
+
+## Introduction
+
+This blog post was written by Jacob Mantooth,Dewan Chowdhury, Arjun Sethi-Olowin and with the help of our advisory Dr.Ruthotto.
```
Comment (lruthotto) [inline content/projects/2022-rl-vs-oc/index.md:14]: 1) Not sure I helped.
2) if you have personal websites, add links!

File: content/projects/2022-rl-vs-oc/index.md [hunk 3]
```diff
@@ -1,9 +1,118 @@
...
...und town is that reinforcement learning is the top dog and has the answers to all our problems. We wanted to see if that really was the case, so this summer we took a trip to Emory University where we looked at the continuous mountain car problem to see if reinforcement learning really, was the best. The continuous mountain car problem is an example of an optimal control problem. You may be asking yourself what is an optimal control problem? An optimal control problem is a dynamical system with an ODE that involves some controllable function $u$. In our case the continuous mountain can be modeled as a PDE where $u$ is some controllable function. In the continuous mountain car problem our control,$u$, is whether the car accelerates left or right. In an optimal control problem, we seek to optimize some objective function, in our case we will minimize the objective function. Our two objective functions are the running cost, which penalizes the car for acceleration. While our other objective function is the terminal cost which penalizes the car for not reaching the goal, the top of the hill, in time. In the image below you can see an example of what the mountain car problem looks like.
```
Comment (lruthotto) [inline content/projects/2022-rl-vs-oc/index.md:18]: I like it! Consider making a few paragraphs (easier to read online)

File: content/projects/2022-rl-vs-oc/index.md [hunk 4]
```diff
@@ -1,9 +1,118 @@
...
+
+
+<p align="center">
+<img src="img/mountaincar.png" width="50%" height="50%"/>
+</p>
+
+### The Why???
```
Comment (lruthotto) [inline content/projects/2022-rl-vs-oc/index.md:25]: Be more specific: Why this problem?

File: content/projects/2022-rl-vs-oc/index.md [hunk 5]
```diff
@@ -1,9 +1,118 @@
...
+
+You may be wondering what our reason was for picking the mountain car problem, there are a couple of reasons why we picked this example. One of the reasons was that the continuous mountain car problem is a well establish benchmark for reinforcement learning models. Another thing that makes the mountain car problem great is that the mountain car problem is in a 2-D state-space which allows for good plotting and visualization. One other reason that we picked the mountain car is that it is both a reinforcement learning and optimal control problem. In our case it has a finite horizon and is in a continuous state. The whole reason we are doing this is because we want to compare three different ways of solving the continuous mountain car problem and see which one really is the best. The three methods that we will be testing can be broken down as one local method and two global methods, we will be discussing each in-detail below.
+
+## Our Three Approaches
+
+### Local
+
+Our first method that we looked at during this REU was the local method. In this approach we found the local solution using numerical ODE solvers and Nonlinear Optimization.
```
Comment (lruthotto) [inline content/projects/2022-rl-vs-oc/index.md:33]: 'a local method'

File: content/projects/2022-rl-vs-oc/index.md [hunk 6]
```diff
@@ -1,9 +1,118 @@
...
+
+# Reinforcement Learning
+
+Our first stop in exploring global methods is reinforcement learning.  We will be using reinforcement learning with actor-critic algorithm. This approach is completely data-based approach. In reinforcement learning it has no knowledge of the model, it only considers the objective function. In reinforcement learning we would like to maximize a reward, so in our case we will maximize negative cost. Reinforcement learning is stochastic in two ways with initial position and action space which allows for exploration. In Reinforcement learning we are trying to estimate an optimal control policy. One of the big things that we have yet to discuss is, what is actor-critic algorithm? In the actor-critic algorithm the critic must learn about and critique whatever policy is currently being followed by the actor. The next picture is a nice visualization of what is going on in the actor-critic algorithm.
+<div style="text-align: center">
+
+![H. Giang, T. Hoan, P. Thanh, and I. Koo. Hybrid noma/oma-based dynamic power allocation scheme using deep reinforcement learning in 5g networks. Applied Sciences, 10(12), 2020.](img/actor-critic.png)
```
Comment (lruthotto) [inline content/projects/2022-rl-vs-oc/index.md:78]: what's the copyright on this? If there is a doubt, better not have it.

File: content/projects/2022-rl-vs-oc/index.md [hunk 7]
```diff
@@ -1,9 +1,118 @@
...
+
+</div>
+
+Some useful background in our reinforcement learning method is that we were working in the OpenAI gym mountain car environment, so we were able to find preexisting code for our project. The thing is our preexisting code was not the same as our problem, so we had to modify it some. The modification we did was the following: we changed the rewards to be continuous and we made it have a fixed final time instead of ending the iteration once reaching the goal. The following video is the results that we were able to get after many training cycles.
```
Comment (lruthotto) [inline content/projects/2022-rl-vs-oc/index.md:82]: could end up in the 'why' section?

File: content/projects/2022-rl-vs-oc/index.md [hunk 8]
```diff
@@ -1,9 +1,118 @@
...
+
+### Week 2 & 3
+
+We decided to spend two weeks to look at reinforcement learning, during these two week we were able to produce a PowerPoint in beamer for our mid-week presentation. In week two and three we looked at our first global method. Dr.Ruthotto handed us some pre written code to play around with. A thing that should be noted is that this prewritten code only worked maybe 50% of the time. Looking deeper into the code we realized that we would have to mess around with the code to get it to match our problem. After we made these couple of changes in our code, we saw how fragile reinforcement learning is, instead of working 50% of the time our code barely worked at all. During this week we were also able to produce a rendering of both of our local and RL methods, adding a nice touch to our presentation that we gave.
+
+### Week 4 & 5
```
Comment (lruthotto) [inline content/projects/2022-rl-vs-oc/index.md:106]: to be completed!

**Review by lruthotto — CHANGES_REQUESTED** [review]
Great first draft, y'all. Please look at my comments inline and push any changes to this branch .

**Review by lruthotto — CHANGES_REQUESTED** [review]
Sorry I didn't submit all the comments for my review.

**Comment by lruthotto** [comment]
Great job, guys. I've merged this now!

**Pull request merged**